File: app/src/main/java/com/vacuum/soleeklab/data/DataManager.java
package com.vacuum.soleeklab.data;

import android.database.Observable;

import com.vacuum.soleeklab.data.prefs.Helper;
import com.vacuum.soleeklab.data.prefs.PreferencesHelper;
import com.vacuum.soleeklab.data.prefs.SharedPrefsHelper;

public interface DataManager extends PreferencesHelper{

    void updateApiHeader(Long userId, String accessToken);

    void setUserAsLoggedOut();

    void updateUserInfo(
            String accessToken,
            Long userId,
            Boolean loggedInMode,
            String userName,
            String email,
            String profilePicPath);

    boolean LoggedInMode();
}

File: app/src/main/java/com/vacuum/soleeklab/data/DataManager.kt
package com.vacuum.mvp3.data

import android.provider.ContactsContract

class DataManager{


    DataManager(SharedPrefsHelper mSharedPrefsHelper){
        mSharedPrefsHelper.this = mSharedPrefsHelper;
    }
    void clear(){
        mSharedPrefsHelper.clear();
    }
    void saveEmailId(Email:String){
        mSharedPrefsHelper.putEmail(ContactsContract.CommonDataKinds.Email);
    }
    String getEmailId() {
        return mSharedPrefsHelper.email;
    }

    void setLoggedIn() {
        mSharedPrefsHelper.loggedInMode = true;
    }

    void getLoggedInMode(): Boolean? {
        return mSharedPrefsHelper.loggedInMode;
    }
}
